src/components/user/user.service.ts

```diff
@@ -209,58 +207,131 @@ export class UserService {
           }),
-          (user)-[:token {active: true, createdAt: datetime()}]->(token),
-          (user)-[:password {active: true, createdAt: datetime()}]->
-          (password:Property {
+        ],
+        [
+          node('user'),
+          relation('out', '', 'token', {
             active: true,
-            value: $pash
+            createdAt: 'datetime()',
           }),
-          (user)-[:realFirstName {active: true, createdAt: datetime()}]->
-          (realFirstName:Property {
+          node('token'),
+        ],
+        [
+          node('user'),
+          relation('out', '', 'password', {
             active: true,
-            value: $realFirstName
+            createdAt: 'datetime()',
           }),
-          (user)-[:realLastName {active: true, createdAt: datetime()}]->
-          (realLastName:Property {
+          node('password', 'Property', {
             active: true,
-            value: $realLastName
+            value: pash,
           }),
-          (user)-[:displayFirstName {active: true, createdAt: datetime()}]->
-          (displayFirstName:Property {
+        ],
+        [
+          node('user'),
+          relation('out', '', 'realFirstName', {
             active: true,
-            value: $displayFirstName
+            createdAt: 'datetime()',
           }),
-          (user)-[:displayLastName {active: true, createdAt: datetime()}]->
-          (displayLastName:Property {
+          node('realFirstName', 'Property', {
             active: true,
-            value: $displayLastName
+            value: input.realFirstName,
           }),
-          (user)-[:phone {active: true, createdAt: datetime()}]->
-          (phone:Property {
+        ],
+        [
+          node('user'),
+          relation('out', '', 'realLastName', {
             active: true,
-            value: $phone
+            createdAt: 'datetime()',
           }),
-          (user)-[:timezone {active: true, createdAt: datetime()}]->
-          (timezone:Property {
+          node('realLastName', 'Property', {
             active: true,
-            value: $timezone
+            value: input.realLastName,
           }),
-          (user)-[:bio {active: true, createdAt: datetime()}]->
-          (bio:Property {
+        ],
+        [
+          node('user'),
+          relation('out', '', 'displayFirstName', {
             active: true,
-            value: $bio
+            createdAt: 'datetime()',
           }),
-          (user)<-[:member]-
-          (acl:ACL {
+          node('displayFirstName', 'Property', {
+            active: true,
+            value: input.displayFirstName,
+          }),
+        ],
+        [
+          node('user'),
+          relation('out', '', 'displayLastName', {
+            active: true,
+            createdAt: 'datetime()',
+          }),
+          node('displayLastName', 'Property', {
+            active: true,
+            value: input.displayLastName,
+          }),
+        ],
```

Comment:
Most of these property queries are the same. How about a helper function?
```ts
const property = (prop: string, value: any) => {
  if (!value) {
    return [];
  }
  return [
    node('user'),
    relation('out', '', prop, {
      active: true,
      createdAt: DateTime.local().toNeo4JDateTime(),
    }),
    node(prop, 'Property', {
      active: true,
      value,
    }),
  ];
};

.create([
  ...
  property('phone', input.phone),
  ...
])
```

Reply:
Love this idea. Should this be an enclosed method you think? Or a private method in the db service class?

Reply:
Changes applied!